app/controllers/sprangular/products_controller.rb

```diff
@@ -1,10 +1,13 @@
 class Sprangular::ProductsController < Sprangular::BaseController
   def index
-    @products = product_scope
-    @products = @products.where("spree_prices.amount IS NOT NULL").where("spree_prices.currency" => current_currency) unless Spree::Config.show_products_without_price
-    @products = @products.ransack(params[:q]).result if params[:q]
-    @products = @products.distinct.page(params[:page]).per(params[:per_page])
-    @cache_key = [I18n.locale, @current_user_roles.include?('admin'), current_currency, params[:q], params[:per_page], params[:per_page]]
+    searcher = Spree::Config.searcher_class.new(params)
+
+    @products = searcher.retrieve_products
+
+    @cache_key = [I18n.locale,
+                  @current_user_roles.include?('admin'),
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.